### PR TITLE
[PLATFORM-922] Add SetPrice component

### PR DESCRIPTION
**Pull request by tumppi**
Added SetPrice component for selecting price for product. Check it out on Storybook -> Marketplace -> SetPrice.

TODOs left:
1. Styling
2. Is `Fix price in fiat` really needed since we have the arrow button to switch base currency? I think it should not be there.

### CODE REVIEW

**Comment by juhah** [comment]
This was recently added to Abstract: https://share.goabstract.com/0a7f8159-7e41-4ee5-a845-4fa0c429019d?mode=design&sha=33fbb87f5e03b79f2d20cdf153c7c5722a760145

Free product should keep the controls but disable and add transparency:
Screen Shot 2019-07-26 at 14 41 28

> Is Fix price in fiat really needed since we have the arrow button to switch base currency? 

I believe it's still needed, the switching inputs allow you to define a value in either currency but the price is kept in DATA. With toggle on, it ties to the USD value.

**Comment by tumppi** [comment]
Ah, didn't notice those designs in time :) I can fix these after my holiday or someone else can take this over.

**Comment by juhah** [comment]
Added beneficiary address & disabling while free product is selected. It updates the `pricePerSecond` value to the product but otherwise selections are kept when toggling between paid / free. Styling is in complete but can be polished later.

File: app/src/marketplace/containers/EditProductPage2/BeneficiaryAddress.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import React, { useCallback } from 'react'
+import cx from 'classnames'
+
+import styles from './BeneficiaryAddress.pcss'
```
Comment (fonty1) [inline app/src/marketplace/containers/EditProductPage2/BeneficiaryAddress.jsx]: lowecase b

File: app/src/marketplace/containers/EditProductPage2/BeneficiaryAddress.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const BeneficiaryAddress = ({ address, onChange, disabled, className }: Props) => {
+    const onAddressChange = useCallback((e: SyntheticInputEvent<EventTarget>) => {
+        onChange(e.target.value)
+    }, [onChange])
+
+    return (
+        <form autoComplete="off">
+            <label className={cx(styles.root, className)}>
+                <strong>Set recipient ETH address</strong>
```
Comment (fonty1) [inline app/src/marketplace/containers/EditProductPage2/BeneficiaryAddress.jsx]: Put 'Set recipient ETH address' in translate file?

File: app/src/marketplace/containers/EditProductPage2/PriceSelector.jsx [hunk 3]
```diff
@@ -1,11 +1,103 @@
...
+
+    const updateRef: Ref<Function> = useRef()
+    updateRef.current = updatePricePerSecond
+
+    const onPriceTypeChange = useCallback((type) => {
+        setIsPaid(type !== 'Free')
+    }, [setIsPaid])
+
+    useEffect(() => {
+        if (updateRef.current) {
```
Comment (fonty1) [inline app/src/marketplace/containers/EditProductPage2/PriceSelector.jsx]: move this into it its own function ?
```
    const calculatePricePerSecond = useCallback(() => {
        if (!isPaid) {
            return BN(0)
        }

        const newPrice = (currency !== currencies.DATA)
            ? convert(price || '0', dataPerUsd, currency, currencies.DATA)
            : price

        return pricePerSecondFromTimeUnit(newPrice || BN(0), timeUnit)
    }, [currency, dataPerUsd, isPaid, price, timeUnit])

...
...
     updateRef.current(calculatePricePerSecond())
```

**Pull request merged**